src/codegen/mod.rs

```diff
@@ -1120,3 +1121,125 @@ impl Expression {
         )
     }
 }
+
+#[derive(PartialEq, Debug, Clone, Copy)]
+pub enum Builtin {
+    Accounts,
+    AddMod,
+    ArrayLength,
+    Balance,
+    Blake2_128,
+    Blake2_256,
+    BlockCoinbase,
+    BlockDifficulty,
+    BlockHash,
+    BlockNumber,
+    Calldata,
+    ExternalFunctionAddress,
+    FunctionSelector,
+    Gasleft,
+    GasLimit,
+    Gasprice,
+    GetAddress,
+    MinimumBalance,
+    MulMod,
+    Keccak256,
+    Origin,
+    Random,
+    ReadAddress,
+    ReadInt8,
+    ReadInt16LE,
+    ReadInt32LE,
+    ReadInt64LE,
+    ReadInt128LE,
+    ReadInt256LE,
+    ReadUint16LE,
+    ReadUint32LE,
+    ReadUint64LE,
+    ReadUint128LE,
+    ReadUint256LE,
+    Ripemd160,
+    Sender,
+    Slot,
+    Sha256,
+    Signature,
+    SignatureVerify,
+    Timestamp,
+    TombstoneDeposit,
+    Value,
+    WriteAddress,
+    WriteInt8,
+    WriteInt16LE,
+    WriteInt32LE,
+    WriteInt64LE,
+    WriteInt128LE,
+    WriteInt256LE,
+    WriteUint16LE,
+    WriteUint32LE,
+    WriteUint64LE,
+    WriteUint128LE,
+    WriteUint256LE,
+}
+
+impl Builtin {
+    pub fn from_ast(ast_builtin: &ast::Builtin) -> Builtin {
```

Comment:
Would it possible to implement the `From` trait here, like:

```
impl From<&ast::Builtin> for Builtin {
...
}
```
So you can use `.into()` to covert into `codegen::Builtin`